### ISSUE TITLE
Support for C# in Rider

### ISSUE THREAD

**Issue by oxysoft** [xylo/intellij-postfix-templates#64]

**Comment by xylo** [comment]
Yeah.  I already tried to add support for C#, because I'm using it, too.  Unfortunately, Rider is completely different from IDEA.  It has nearly nothing in common with IDEA, because Rider uses the Intellij Platform only as a wrapper for ReSharper.  See https://blog.jetbrains.com/dotnet/2017/02/07/rider-front-end-plugin-development/ for details.

It's really a pity!  In order to implement this plugin for Rider one has to reimplement the whole plugin in C# for ReSharper.  Unfortunately, I don't have the time for that and I also don't want to maintain two different code basis.
So as long as JetBrains does this trick/workaround with ReSharper I don't see that Custom Postfix Templates in coming to Rider.  IMO, Rider is by far better than Visual Studio.  So I hope that more and more C# developers switch to Rider.  And maybe at some point (when the market share of Visual Studio dropped below 30%) JetBrains will give up ReSharper and reimplement the whole thing with the Intellij Platform.  That would be great!  Then Rider could benefit from a lot plugins that are currently incompatible with it.  And I would also start adding support for C#/Rider for this plugin.

**Comment by xylo** [comment]
Today I learned that ReSharper 9.1 comes with a feature similar to "postfix templates".  It's called "source templates" and can be configured by the user: https://www.jetbrains.com/help/resharper/Source_Templates.html

**Comment by xylo** [comment]
Status update: According to https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000670130/comments/360000161480 there is a simple C# parser coming in Rider 2018.3 EAP2.
I will take a look at it in the next days it to see if it is sufficient.

**Comment by w371987114** [comment]
> Status update: According to https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000670130/comments/360000161480 there is a simple C# parser coming in Rider 2018.3 EAP2.
> I will take a look at it in the next days it to see if it is sufficient.

Is it means the extension will supported in visual studio and c#?

**Comment by LaP0573** [comment]
> Rider consists of a small Intellij IDEA side and a plugin called ReSharper. ReSharper is also a plugin for Visual Studio to provide a nice code editor. My plugin is implemented in Java (for Intellij IDEA). So it cannot be used in ReSharper/Visual Studio.
> On the weekend I checked whether I can add support for C# in Rider, but so far it did not work. I already asked a JetBrains developer what I can do, but did not get a response yet.

Any news about this? I use CPT extensively with IntelliJ in Java and it would be incredible to also be able to use it with Rider!

**Comment by xylo** [comment]
No changes since 2018. :\
